src/storage/log_page_async_fetcher.cpp

```diff
@@ -0,0 +1,60 @@
+#include "log_page_async_fetcher.hpp"
+
+#include "log_impl.h"
+#include "thread_manager.hpp"
+
+namespace cublog
+{
+  class log_page_fetch_task : public cubthread::entry_task
+  {
+    public:
+      explicit log_page_fetch_task (
+	      LOG_PAGEID pageid,
+	      async_page_fetcher::callback_func_type &&callback)
+	:
+	cubthread::entry_task (),
+	m_logpageid (pageid),
+	m_callback (std::move (callback))
+      {
+      }
+
+      void execute (context_type &) override
```

Comment:
I recommend separating class declaration of function definitions. It is easier to visualize the entire class interface on its whole and understand its purpose.

I suggest moving log_page_fetch_task::execute definition outside the class.